internal/sidecar/service/volumereplication.go

```diff
@@ -290,3 +257,28 @@ func (rs *ReplicationServer) GetVolumeReplicationInfo(
 		LastSyncBytes:    resp.GetLastSyncBytes(),
 	}, nil
 }
+
+// setReplicationSource sets the replication source for the given ReplicationSource.
+func setReplicationSource(src *csiReplication.ReplicationSource, req *proto.ReplicationSource) error {
+	if src == nil {
+		src = &csiReplication.ReplicationSource{}
+	}
+
+	switch {
+	case req == nil:
+		return errors.New("replication source is required")
+	case req.GetVolume() == nil && req.GetVolumegroup() == nil:
```

Comment:
it looks like we missed something somewhere else, `GetVolumegroup()` should have been `GetVolumeGroup()`

Reply:
it was a typo in proto file itself, its updated to use volume_group instead of volumegroup